compiler/noirc_evaluator/src/ssa/opt/inlining.rs

```diff
@@ -98,20 +98,22 @@ struct InlineContext {
 
     // These are the functions of the program that we shouldn't inline.
     functions_not_to_inline: BTreeSet<FunctionId>,
+
+    globals: &'global Function,
 }
 
 /// The per-function inlining context contains information that is only valid for one function.
 /// For example, each function has its own DataFlowGraph, and thus each function needs a translation
 /// layer to translate between BlockId to BlockId for the current function and the function to
 /// inline into. The same goes for ValueIds, InstructionIds, and for storing other data like
 /// parameter to argument mappings.
-struct PerFunctionContext<'function> {
+struct PerFunctionContext<'function, 'global> {
```

Comment:
I'm unsure that we actually need two lifetimes here. Can we make do with one?

Reply:
Yeah good call. I can pass the globals to the `FunctionContext` directly.